### PR TITLE
Add napalm login details to Router edit Form, not configuration.py

### PR DESCRIPTION
**Pull request by smaxx1337**
With this PR Napalm login details are not put into the configuration.py anymore, they are getting put in the Router form. It works with optional_args, but since I have no device that requires no optional_args for a connection I can't test if it actually works when someone doesn't put anything in the optional_args field. So it would be cool if you could test if that works too.

Any feedback is appreciated.

### CODE REVIEW

File: peering/models.py [hunk 1]
```diff
@@ -1357,10 +1365,10 @@ def get_napalm_device(self):
             self.logger.debug('found napalm driver "%s"', self.platform)
             return driver(
                 hostname=self.hostname,
-                username=settings.NAPALM_USERNAME,
-                password=settings.NAPALM_PASSWORD,
-                timeout=settings.NAPALM_TIMEOUT,
-                optional_args=settings.NAPALM_ARGS,
+                username=self.napalm_username,
```
Comment (respawner) [inline peering/models.py]: Continue to use the existing configuration options if the user does not define a per-router NAPALM configuration.

File: peering/tables.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ class BGPSessionStateColumn(tables.TemplateColumn):
     def __init__(self, *args, **kwargs):
         default = kwargs.pop("default", "")
         visible = kwargs.pop(
-            "visible", settings.NAPALM_USERNAME and settings.NAPALM_PASSWORD
+            "visible", Router.napalm_username and Router.napalm_password
```
Comment (respawner) [inline peering/tables.py]: This won't work. We need to perform this check against a router instance.

File: peering_manager/settings.py [hunk 3]
```diff
@@ -88,10 +88,6 @@
 LOGGING = getattr(configuration, "LOGGING", DEFAULT_LOGGING)
 CHANGELOG_RETENTION = getattr(configuration, "CHANGELOG_RETENTION", 90)
 LOGIN_REQUIRED = getattr(configuration, "LOGIN_REQUIRED", False)
-NAPALM_USERNAME = getattr(configuration, "NAPALM_USERNAME", "")
```
Comment (respawner) [inline peering_manager/settings.py]: Keep current configuration options for people who still want to use them.

File: templates/peering/router/add_edit.html [hunk 4]
```diff
@@ -13,6 +13,10 @@
                 {% render_field form.platform %}
                 {% render_field form.encrypt_passwords %}
                 {% render_field form.configuration_template %}
+                {% render_field form.napalm_username %}
```
Comment (respawner) [inline templates/peering/router/add_edit.html]: Put this inside a dedicated Bootstrap `card` separate these form fields from the others.

File: peering/forms.py [hunk 5]
```diff
@@ -707,6 +707,11 @@ class RouterForm(BootstrapMixin, forms.ModelForm):
     )
     comments = CommentField()
     tags = TagField(required=False)
+    
+    napalm_username = forms.CharField(required=False, label="Username")
+    napalm_password = forms.CharField(required=False, label="Password")
+    napalm_timeout = forms.IntegerField(label="Timeout in seconds", initial=0)
```
Comment (respawner) [inline peering/forms.py]: Mark this field as not required, set its default value to 30 seconds like defined in the model. Call it just `Timeout` and set a helper text to indicate what it is for and what is its unit.

File: peering/forms.py [hunk 6]
```diff
@@ -707,6 +707,11 @@ class RouterForm(BootstrapMixin, forms.ModelForm):
...
+    napalm_args = forms.CharField(required=False, label="Optional Arguments (dictionary)")
```
Comment (respawner) [inline peering/forms.py]: It could be useful to have a link to the NAPALM documentation for this field inside its helper text.

Comment (smaxx1337) [reply peering/forms.py]: See the help_texts dictionary, but i don't know how to "activate" them. If you tell me how to do that, I will do that.

Comment (smaxx1337) [reply peering/forms.py]: Could you gvie me a documentation link or a example in your code where you display the help text?

Comment (respawner) [reply peering/forms.py]: Help texts for form fields can be defined in the `help_texts` dictionary of form `Meta` classes. They can also be defined right in the field attribute of a model by using the `help_text` named parameter of a field constructor.

Comment (smaxx1337) [reply peering/forms.py]: What I have tried is:
`napalm_timeout = models.PositiveIntegerField(blank=True, default=30, help_text="Test")`
in models.py, made a migration (it altered the field), but nothing changed.
And then I've tried adding it to the help_text dict in forms.py:```

    class Meta:
        model = Router

        fields = (
            "netbox_device_id",
            "use_netbox",
            "name",
            "hostname",
            "napalm_username",
            "napalm_password",
            "napalm_timeout",
            "napalm_args",
            "platform",
            "encrypt_passwords",
            "configuration_template",
            "comments",
            "tags",
        )
        labels = {"use_netbox": "Use NetBox", "configuration_template": "Configuration"}
        help_texts = {
            "hostname": "Router hostname (must be resolvable) or IP address",
            "configuration_template": "Template used to generate device configuration",
            "napalm_timeout": "Maximum time to wait for a connection (in seconds)",
            "napalm_args": "see .. for more",
```
Still no help_text.
Im kinda stuck now... 😅

File: peering/tables.py (hunk 2 above)
Comment (smaxx1337) [reply peering/tables.py]: I have changed it with a if, but i'm quite not sure if that will work.

**Comment by mirackle-spb** [comment]
Little bump here. This is needed feature!

**Comment by respawner** [comment]
I've pushed some of your changes in c3cab43 and adapted some others based on the latest master branch. Thanks for working on this PR, we can close it now.

**Pull request closed without merging**